src/commands/stepfunctions/instrument.ts

```diff
@@ -350,15 +351,10 @@ export class InstrumentStepFunctionsCommand extends Command {
         hasChanges = true
       }
 
-      if (this.mergeStepFunctionAndLambdaTraces) {
+      if (this.mergeStepFunctionAndLambdaTraces || this.propagateUpstreamTrace) {
```

Comment:
So, if either one of the flag is used, the nested SF definition is overwritten, and there are 3 possible case
Lambda: Yes, Nested: Yes
Lambda: No, Nested: No
Lambda: No, Nested: Yes.
-------
BTW: Do we really need to have an extra flag to support the case of Lambda No, Nested Yes? What about we reuse the current flag "mergeStepFunctionAndLambdaTraces", and once it is used by the customer, both definitions of Lambda and nested SF are modified.  @sumedha.

Reply:
I'm fine reusing `mergeStepFunctionAndLambdaTraces` for both. We discussed this in #serverlesstraceandspanlinking, but didn't really come to a consensus. 

> Chris Agocs
>   12:25
> @sumedha @kimi What do we think of --trace-nested-step-function-invocations as a flag to pass to datadog ci to toggle passing context between nested step function invocations? (edited) 
> 
> Chris Agocs
>   14:04
> @sumedha, the other option is, do we want to make it one flag that controls both Stepfunction -> Lambda and Stepfunction -> Stepfunction? Thoughts? (edited) 
> 
> 
> Sumedha Mehta
>   14:53
> like --propogate-upstream-trace ? (edited) 
> 14:54
> I don't see why a customer would want to turn it on and off per-service
> 
> 
> Kimi Wu
>   14:58
> Are you proposing a new flag to replace the old one? I think we still need to keep the old one working if that's the suggestion though.
> 
> 
> Chris Agocs
>   15:09
> I'd keep the old flag as well to preserve backward compatibility. I think it would be safe for --merge-step-function-and-lambda-traces and --propogate-upstream-trace to do the same thing, and turn on trace propagation from stepfunction->lambda and stepfunction->stepfunction (edited)